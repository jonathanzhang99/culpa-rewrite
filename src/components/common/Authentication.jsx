import PropTypes from "prop-types";
import React, { useContext, useReducer } from "react";
import { Route, Redirect } from "react-router-dom";

export const AuthContext = React.createContext(null);

function authReducer(state, action) {
  switch (action.type) {
    case "LOGIN_START":
      return {
        ...state,
        isLoading: true,
      };
    case "LOGIN_ERROR":
      return {
        ...state,
        isError: true,
        isLoading: false,
      };

    case "LOGIN_SUCCESS":
      localStorage.setItem("authToken", JSON.stringify(action.payload));
      return {
        ...state,
        isError: false,
        isLoading: false,
        isLoggedIn: true,
        authToken: action.payload,
      };

    case "LOGIN_FAILURE":
      return {
        ...state,
        isError: false,
        isLoading: false,
        isLoggedIn: false,
      };
    case "LOGOUT_SUCCESS":
      localStorage.removeItem("authToken");
      return {
        ...state,
        isError: false,
        isLoading: false,
        isLoggedIn: false,
        authToken: undefined,
      };
    default:
      throw new Error();
  }
}

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]).isRequired,
};

/**
 * AuthProvider createa React Context with values such that children elements
 * in the tree can access user auth token and manipulate them
 * (e.g. login, logout)
 *
 */
export function AuthProvider({ children }) {
  const existingToken = JSON.parse(localStorage.getItem("authToken"));

  const [state, dispatch] = useReducer(authReducer, {
    isError: false,
    isLoading: false,
    isLoggedIn: !!existingToken,
    authToken: existingToken,
  });

  // `login` either returns an error or null if a token was succesfully
  // retrieved. The error will always be a string message
  const login = async ({ username, password }) => {
    dispatch({ type: "LOGIN_START" });
    const response = await fetch("/api/auth/login", {
      method: "POST",
      body: JSON.stringify({
        username,
        password,
      }),
      headers: {
        "Content-Type": "application/json",
      },
    });
    try {
      const result = await response.json();

      // Require that the backend sends either a token or an error
      result.error
        ? dispatch({ type: "LOGIN_FAILURE" })
        : dispatch({ type: "LOGIN_SUCCESS", payload: result.token });
      return result;
    } catch (err) {
      dispatch({ type: "LOGIN_ERROR" });
      return err;
    }
  };

  // `logout` simply removes the token from the client. It does NOT
  // expire the token.
  const logout = async () => {
    if (state.authToken) {
      dispatch({ type: "LOGOUT_SUCCESS" });
    }
  };

  return (
    <AuthContext.Provider
      value={{
        ...state,
        login,
        logout,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
}

AuthProvider.propTypes = propTypes;

/**
 * ProtectedRoute allows access to the url only if a user is logged in.
 *
 */
export function ProtectedRoute(props) {
  const { isLoggedIn } = useContext(AuthContext);
  // eslint-disable-next-line react/jsx-props-no-spreading
  return isLoggedIn ? <Route {...props} /> : <Redirect to="/" />;
}
