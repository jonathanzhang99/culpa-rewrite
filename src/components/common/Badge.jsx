import PropTypes from "prop-types";
import React from "react";
import { Image } from "semantic-ui-react";

import bronzeNugget from "icons/bronze-nugget.svg";
import goldNugget from "icons/gold-nugget.svg";
import silverNugget from "icons/silver-nugget.svg";

const propTypesBadge = {
  badgeId: PropTypes.number.isRequired,
};

export default function Badge({ badgeId }) {
  switch (badgeId) {
    case 1: {
      return <Image avatar spaced="left" src={goldNugget} />;
    }
    case 2: {
      return <Image avatar spaced="left" src={silverNugget} />;
    }
    case 3: {
      return <Image avatar spaced="left" src={bronzeNugget} />;
    }
    default: {
      return null;
    }
  }
}

Badge.propTypes = propTypesBadge;
