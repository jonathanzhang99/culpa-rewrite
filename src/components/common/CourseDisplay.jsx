import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";
import { Header } from "semantic-ui-react";

const propTypesCourseDisplayName = {
  as: PropTypes.string,
  name: PropTypes.string.isRequired,
  code: PropTypes.string,
};

const propTypesCourseDisplayLink = {
  as: PropTypes.string,
  courseId: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  code: PropTypes.string,
};

const defaultProps = {
  as: "p",
  code: "",
};

export function CourseDisplayName({ as, code, name }) {
  return (
    <Header as={as}>
      {code} {name}
    </Header>
  );
}

export function CourseDisplayLink({ as, courseId, code, name }) {
  return (
    <Link
      to={{
        pathname: `/courses/${courseId}`,
      }}
    >
      <CourseDisplayName as={as} code={code} name={name} />
    </Link>
  );
}

CourseDisplayName.propTypes = propTypesCourseDisplayName;
CourseDisplayName.defaultProps = defaultProps;

CourseDisplayLink.propTypes = propTypesCourseDisplayLink;
CourseDisplayLink.defaultProps = defaultProps;
