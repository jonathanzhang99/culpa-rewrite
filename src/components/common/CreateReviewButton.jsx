/* eslint-disable react/jsx-props-no-spreading */
import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";
import { Button } from "semantic-ui-react";

const propTypes = {
  courseId: PropTypes.number,
  professorId: PropTypes.number,
};

const defaultProps = {
  courseId: undefined,
  professorId: undefined,
};

export default function CreateReviewButton({ courseId, professorId, ...rest }) {
  return (
    <Link
      to={{
        pathname: "/review/submit",
        state: { courseId, professorId },
      }}
    >
      <Button {...rest} />
    </Link>
  );
}

CreateReviewButton.propTypes = propTypes;
CreateReviewButton.defaultProps = defaultProps;
