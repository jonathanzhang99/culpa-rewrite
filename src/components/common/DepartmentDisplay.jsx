import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";
import { Header } from "semantic-ui-react";

const propTypesDepartmentDisplayName = {
  as: PropTypes.oneOf(["header", "text"]),
  departmentName: PropTypes.string.isRequired,
};

const propTypesDepartmentDisplayLink = {
  as: PropTypes.oneOf(["header", "text"]),
  departmentId: PropTypes.number.isRequired,
  departmentName: PropTypes.string.isRequired,
};

const defaultProps = {
  as: "text",
};

export function DepartmentDisplayName({ as, departmentName }) {
  const DepartmentName = as === "header" ? Header : "span";
  return <DepartmentName> {departmentName} </DepartmentName>;
}

export function DepartmentDisplayLink({ as, departmentId, departmentName }) {
  return (
    <Link to={`/department/${departmentId}`}>
      <DepartmentDisplayName as={as} departmentName={departmentName} />
    </Link>
  );
}

DepartmentDisplayName.propTypes = propTypesDepartmentDisplayName;
DepartmentDisplayName.defaultProps = defaultProps;

DepartmentDisplayLink.propTypes = propTypesDepartmentDisplayLink;
DepartmentDisplayLink.defaultProps = defaultProps;
