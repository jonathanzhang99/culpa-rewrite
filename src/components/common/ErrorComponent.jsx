import React from "react";

export default function ErrorComponent() {
  return <div>error occurred</div>;
}
