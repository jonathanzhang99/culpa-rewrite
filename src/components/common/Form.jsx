import PropTypes from "prop-types";
import React from "react";
import { useForm } from "react-hook-form";
import { Form as SemanticForm, Message } from "semantic-ui-react";

const FORM_ERRORS = "form";

const propTypes = {
  defaultValues: PropTypes.objectOf(PropTypes.any),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
  ]),
  onSubmit: PropTypes.func.isRequired,
  onSuccess: PropTypes.func.isRequired,
};

const defaultProps = {
  defaultValues: {},
  children: [],
};

export default function Form({ defaultValues, children, onSubmit, onSuccess }) {
  const { handleSubmit, register, errors, setError, clearErrors } = useForm({
    defaultValues,
  });

  const onSubmitWithHandlers = async (data) => {
    const submissionError = await onSubmit(data);
    if (submissionError) {
      setError(FORM_ERRORS, { type: "server", message: submissionError });
    } else {
      onSuccess(data);
    }
  };

  const handleSubmitWithErrors = async () => {
    clearErrors(FORM_ERRORS);
    await handleSubmit(onSubmitWithHandlers)();
  };

  const registeredChildren = React.Children.map(children, (child) => {
    const childError = errors[child.props.name];

    let childErrorContent;
    if (childError) {
      childErrorContent = { content: childError.message };
    }

    if (React.isValidElement(child)) {
      return React.cloneElement(child, {
        register,
        key: child.props.id,
        error: childErrorContent,
      });
    }
    return child;
  });

  return (
    <SemanticForm error={!!errors.form} onSubmit={handleSubmitWithErrors}>
      <Message error header="Login Error" content={errors?.form?.message} />
      {registeredChildren}
    </SemanticForm>
  );
}

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;
