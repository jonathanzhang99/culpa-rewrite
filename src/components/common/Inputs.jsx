/**
 * This file contains all of the input components that can be used with
 * the custom Form component. All components use Semantic UI React and
 * are compatible with react-hook-forms.
 *
 * Many of the prop types are duplicated on purpose in order to give a
 * consistent description of the props that each component accepts.
 *
 * NOTE: These components are NOT generic components. Only a selection
 * of the usable props are passed down to the base Semantic components
 * in order to limit the freedom and chance of breaking changes. If
 * you need additional functionality you will need to pass those props
 * down.
 *
 * NOTE: onChange is passed from Form component via the react-hook-form
 * controller. You should never directly pass in the onChange prop.
 */
/* eslint-disable react/jsx-props-no-spreading */
import debounce from "lodash.debounce";
import PropTypes from "prop-types";
import React, { useReducer, useState } from "react";
import {
  Form as SemanticForm,
  Button,
  Input,
  TextArea,
  Search,
  Dropdown,
  Radio,
  Modal,
} from "semantic-ui-react";

import { FormGroup } from "components/common/Form";

function getId(id, name) {
  return id || `form-input-${name}`;
}

const propTypesInput = {
  error: PropTypes.shape({
    message: PropTypes.string,
  }),
  id: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string.isRequired,
  width: PropTypes.number,
  onChange: PropTypes.func,
};

const defaultPropsInput = {
  error: undefined,
  id: "",
  label: "",
  width: undefined,
  onChange: () => {},
};

export function TextInput({ error, id, label, name, width, onChange }) {
  return (
    <SemanticForm.Field
      aria-label={label}
      control={Input}
      error={error}
      id={getId(id, name)}
      label={label}
      name={name}
      type="text"
      width={width}
      onChange={onChange}
    />
  );
}

export function PasswordInput({ name, label, error, id, width, onChange }) {
  return (
    <SemanticForm.Field
      aria-label={label}
      control={Input}
      error={error}
      id={getId(id, name)}
      label={label}
      name={name}
      type="password"
      width={width}
      onChange={onChange}
    />
  );
}

const propTypesTextAreaInput = {
  error: PropTypes.shape({
    message: PropTypes.string,
  }),
  id: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string.isRequired,
  width: PropTypes.number,
  onChange: PropTypes.func,
  rows: PropTypes.number,
};

const defaultPropsTextAreaInput = {
  error: undefined,
  id: "",
  label: "",
  width: undefined,
  onChange: () => {},
  rows: 3,
};

export function TextAreaInput({
  name,
  label,
  error,
  id,
  width,
  onChange,
  rows,
}) {
  return (
    <SemanticForm.Field
      aria-label={label}
      control={TextArea}
      error={error}
      id={getId(id, name)}
      label={label}
      name={name}
      rows={rows}
      width={width}
      onChange={onChange}
    />
  );
}

const propTypesDropdown = {
  disabled: PropTypes.bool,
  options: PropTypes.arrayOf(PropTypes.any),
  placeholder: PropTypes.string,
  error: PropTypes.shape({
    message: PropTypes.string,
  }),
  id: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string.isRequired,
  width: PropTypes.number,
  onChange: PropTypes.func,
};

const defaultPropsDropdown = {
  disabled: false,
  options: [],
  placeholder: undefined,
  error: undefined,
  id: "",
  label: "",
  width: undefined,
  onChange: () => {},
};
export function DropdownInput({
  disabled,
  name,
  label,
  error,
  id,
  width,
  options,
  placeholder,
  onChange,
}) {
  return (
    <SemanticForm.Field
      search
      selection
      aria-label={label}
      control={Dropdown}
      disabled={disabled}
      error={error}
      id={getId(id, name)}
      label={label}
      name={name}
      options={options}
      placeholder={placeholder}
      width={width}
      onChange={(e, { value }) => onChange(value)}
    />
  );
}

const propTypesRadioInputGroup = {
  name: PropTypes.string.isRequired,
  labels: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      key: PropTypes.any,
    })
  ).isRequired,
  onChange: PropTypes.func,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

const defaultPropsRadioInputGroup = {
  onChange: () => {},
  value: undefined,
};

export function RadioInputGroup({ name, labels, onChange, value }) {
  const radioButtons = labels.map(({ label, key }) => {
    return (
      <SemanticForm.Field
        checked={value === key}
        control={Radio}
        id={`${name}-radio-button-${key}`}
        key={label}
        label={label}
        name={name}
        type="radio"
        value={key}
        onChange={(e, { value: checkedValue }) => onChange(checkedValue)}
      />
    );
  });
  return (
    <FormGroup inline unstackable widths="equal">
      {radioButtons}
    </FormGroup>
  );
}

function searchReducer(state, action) {
  switch (action.type) {
    case "SEARCH_START": {
      return {
        ...state,
        isLoading: true,
      };
    }
    case "SEARCH_SUCCESS": {
      return {
        isLoading: false,
        results: action.payload,
      };
    }
    case "SEARCH_RESET": {
      return {
        isLoading: false,
        results: [],
      };
    }
    case "SEARCH_ERROR": {
      return {
        ...state,
        isLoading: false,
      };
    }
    default:
      throw new Error();
  }
}

const propTypesSearchInput = {
  error: PropTypes.shape({
    message: PropTypes.string,
  }),
  id: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string.isRequired,
  searchEntity: PropTypes.oneOf(["all", "professors", "courses"]),
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  width: PropTypes.number,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  onResultSelect: PropTypes.func,
  onSearchChange: PropTypes.func,
};

const defaultPropsSearchInput = {
  error: undefined,
  id: undefined,
  label: undefined,
  searchEntity: "all",
  onChange: () => {},
  onBlur: () => {},
  onResultSelect: () => {},
  onSearchChange: () => {},
  value: "",
  width: undefined,
};

/**
 * Search bar with autocompletion that receives onChange, onBlur, value from
 * react-hook-forms.
 *
 * onSearchChange: (value) => void
 */
export function SearchInput({
  error,
  id,
  label,
  name,
  searchEntity,
  value,
  width,
  onChange,
  onBlur,
  onResultSelect,
  onSearchChange,
}) {
  const initialState = {
    isLoading: false,
    results: [],
  };
  const [{ isLoading, results }, dispatch] = useReducer(
    searchReducer,
    initialState
  );

  const noResultsMessage = `No ${
    searchEntity === "all" ? "results" : searchEntity
  } found`;

  const handleResultSelect = (e, { result }) => {
    onChange(result);
    onResultSelect(result);
  };

  const handleSearchChange = async (e, { value: searchValue }) => {
    onChange(e);
    onSearchChange(searchValue);

    if (searchValue.length < 2) {
      return dispatch({ type: "SEARCH_RESET" });
    }

    dispatch({ type: "SEARCH_START" });
    const response = await fetch(
      `/api/search?searchEntity=${searchEntity}&query=${searchValue}`,
      {
        method: "GET",
        headers: { "Content-Type": "Application/json" },
      }
    );

    try {
      const result = await response.json();

      if (response.ok) {
        dispatch({ type: "SEARCH_SUCCESS", payload: result.searchResults });
      }
    } catch (err) {
      dispatch({ type: "SEARCH_ERROR" });
    }
    return null;
  };

  return (
    <SemanticForm.Field
      control={Search}
      error={error}
      id={getId(id, name)}
      label={label}
      loading={isLoading}
      name={name}
      noResultsMessage={noResultsMessage}
      results={results}
      value={value?.title || value}
      width={width}
      onBlur={onBlur}
      onResultSelect={handleResultSelect}
      onSearchChange={debounce(handleSearchChange, 300, { leading: true })}
    />
  );
}

export function Submit(props) {
  return (
    <Button name="submit" type="submit" {...props}>
      SUBMIT
    </Button>
  );
}

const propTypesSubmitConfirm = {
  trigger: PropTypes.func,
  handleSubmit: PropTypes.func,
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  header: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
};

const defaultPropsSubmitConfirm = {
  trigger: () => {},
  handleSubmit: () => {},
  content: "",
  header: "",
};

export function SubmitConfirm({ trigger, handleSubmit, content, header }) {
  const [confirmOpen, setConfirmOpen] = useState(false);

  const openConfirm = async () => {
    if (!confirmOpen) {
      const validated = await trigger();
      if (validated) setConfirmOpen(true);
    }
  };

  const closeConfirm = () => setConfirmOpen(false);

  return (
    <>
      <Button name="confirmModal" type="button" onClick={openConfirm}>
        SUBMIT
      </Button>
      <Modal open={confirmOpen} onClose={closeConfirm} onOpen={openConfirm}>
        <Modal.Header>{header}</Modal.Header>
        <Modal.Content>
          <Modal.Description>{content}</Modal.Description>
        </Modal.Content>
        <Modal.Actions>
          {/* TODO: Currently gives no options to specify custom buttons/text */}
          <Button onClick={closeConfirm}>Edit Review</Button>
          <Submit onClick={handleSubmit} />
        </Modal.Actions>
      </Modal>
    </>
  );
}

TextInput.propTypes = propTypesInput;
TextInput.defaultProps = defaultPropsInput;

PasswordInput.propTypes = propTypesInput;
PasswordInput.defaultProps = defaultPropsInput;

TextAreaInput.propTypes = propTypesTextAreaInput;
TextAreaInput.defaultProps = defaultPropsTextAreaInput;

DropdownInput.propTypes = propTypesDropdown;
DropdownInput.defaultProps = defaultPropsDropdown;

RadioInputGroup.propTypes = propTypesRadioInputGroup;
RadioInputGroup.defaultProps = defaultPropsRadioInputGroup;

SearchInput.propTypes = propTypesSearchInput;
SearchInput.defaultProps = defaultPropsSearchInput;

SubmitConfirm.propTypes = propTypesSubmitConfirm;
SubmitConfirm.defaultProps = defaultPropsSubmitConfirm;
