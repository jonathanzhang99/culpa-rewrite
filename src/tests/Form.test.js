import { render } from "@testing-library/react";
import React from "react";

import Form, { Submit } from "components/common/Form";
import { TextInput } from "components/common/Inputs";

describe("Form Component Tests", () => {
  const onSubmit = jest.fn();
  const onSuccess = jest.fn();

  test("empty form", () => {
    const snapshot = render(<Form onSubmit={onSubmit} onSuccess={onSuccess} />);
    expect(snapshot).toMatchSnapshot();
  });

  test("form with single input", () => {
    const snapshot = render(
      <Form onSubmit={onSubmit} onSuccess={onSuccess}>
        <TextInput name="test1" />
      </Form>
    );
    expect(snapshot).toMatchSnapshot();
  });

  test("form with multiple inputs", () => {
    const snapshot = render(
      <Form onSubmit={onSubmit} onSuccess={onSuccess}>
        <TextInput name="test1" />
        <TextInput name="test2" />
      </Form>
    );
    expect(snapshot).toMatchSnapshot();
  });

  test("form with submit button", () => {
    const snapshot = render(
      <Form onSubmit={onSubmit} onSuccess={onSuccess}>
        <TextInput name="test1" />
        <Submit value="submit" />
      </Form>
    );
    expect(snapshot).toMatchSnapshot();
  });
});
