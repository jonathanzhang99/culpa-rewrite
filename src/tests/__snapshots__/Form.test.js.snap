// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component Tests empty form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="ui form"
      >
        <div
          class="ui error message"
        >
          <div
            class="content"
          >
            <div
              class="header"
            >
              Login Error
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="ui form"
    >
      <div
        class="ui error message"
      >
        <div
          class="content"
        >
          <div
            class="header"
          >
            Login Error
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Form Component Tests form with multiple inputs 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="ui form"
      >
        <div
          class="ui error message"
        >
          <div
            class="content"
          >
            <div
              class="header"
            >
              Login Error
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <label
            for="form-input-test1"
          />
          <input
            aria-label=""
            id="form-input-test1"
            name="test1"
            type="text"
          />
        </div>
        <div
          class="field"
        >
          <label
            for="form-input-test2"
          />
          <input
            aria-label=""
            id="form-input-test2"
            name="test2"
            type="text"
          />
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="ui form"
    >
      <div
        class="ui error message"
      >
        <div
          class="content"
        >
          <div
            class="header"
          >
            Login Error
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <label
          for="form-input-test1"
        />
        <input
          aria-label=""
          id="form-input-test1"
          name="test1"
          type="text"
        />
      </div>
      <div
        class="field"
      >
        <label
          for="form-input-test2"
        />
        <input
          aria-label=""
          id="form-input-test2"
          name="test2"
          type="text"
        />
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Form Component Tests form with single input 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="ui form"
      >
        <div
          class="ui error message"
        >
          <div
            class="content"
          >
            <div
              class="header"
            >
              Login Error
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <label
            for="form-input-test1"
          />
          <input
            aria-label=""
            id="form-input-test1"
            name="test1"
            type="text"
          />
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="ui form"
    >
      <div
        class="ui error message"
      >
        <div
          class="content"
        >
          <div
            class="header"
          >
            Login Error
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <label
          for="form-input-test1"
        />
        <input
          aria-label=""
          id="form-input-test1"
          name="test1"
          type="text"
        />
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Form Component Tests form with submit button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="ui form"
      >
        <div
          class="ui error message"
        >
          <div
            class="content"
          >
            <div
              class="header"
            >
              Login Error
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <label
            for="form-input-test1"
          />
          <input
            aria-label=""
            id="form-input-test1"
            name="test1"
            type="text"
          />
        </div>
        <button
          class="ui button"
          name="submit"
          type="submit"
          value="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="ui form"
    >
      <div
        class="ui error message"
      >
        <div
          class="content"
        >
          <div
            class="header"
          >
            Login Error
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <label
          for="form-input-test1"
        />
        <input
          aria-label=""
          id="form-input-test1"
          name="test1"
          type="text"
        />
      </div>
      <button
        class="ui button"
        name="submit"
        type="submit"
        value="submit"
      >
        Submit
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
